examples/cndpfwd/hs-fwd.c

```diff
@@ -0,0 +1,258 @@
+                    thd_private->scratch, hsfwd_handler, &dst_lport) != HS_SUCCESS)
+            return -1;
+
+        if (dst_lport & 0x8000) {
+            pktmbuf_free(pkt);
+            continue;
+        }
+
+        dst = jcfg_lport_by_index(fwd->jinfo, (dst_lport & 0xFF));
+        if (!dst)
+            /* Cannot forward to non-existing port, so echo back on incoming interface */
+            dst = lport;
+
+        MAC_SWAP(pktmbuf_mtod(pkt, void *));
+        (void)txbuff_add(txbuff[dst->lpid], pkt);
+    }
+
+    int nb_lports = jcfg_num_lports(fwd->jinfo);
+    for (int i = 0; i < nb_lports; i++) {
+        jcfg_lport_t *dst = jcfg_lport_by_index(fwd->jinfo, i);
+
+        if (!dst)
+            continue;
+
+        /* Could hang here if we can never flush the TX packets */
+        while (txbuff_count(txbuff[dst->lpid]) > 0)
+            txbuff_flush(txbuff[dst->lpid]);
+    }
+
+    return 0;
+#else
+    CNE_SET_USED(lport);
+    CNE_SET_USED(fwd);
+    return -1;
+#endif /* ENABLE HYPERSCAN */
+}
+
+#ifdef ENABLE_HYPERSCAN
+static unsigned
+hs_parse_flags(const char *str)
+{
+    unsigned flags = 0;
+
+    if (!str)
+        return flags;
+
+    for (; *str != '\0'; str++) {
+        switch (*str) {
+        case 'i':
+            flags |= HS_FLAG_CASELESS;
+            break;
+        case 'm':
+            flags |= HS_FLAG_MULTILINE;
+            break;
+        case 's':
+            flags |= HS_FLAG_DOTALL;
+            break;
+        case 'H':
+            flags |= HS_FLAG_SINGLEMATCH;
+            break;
+        case 'V':
+            flags |= HS_FLAG_ALLOWEMPTY;
+            break;
+        case '8':
+            flags |= HS_FLAG_UTF8;
+            break;
+        case 'W':
+            flags |= HS_FLAG_UCP;
+            break;
+        case '\r':        // stray carriage-return
+            break;
+        case '\n':        // stray newline
+            break;
+        default:
+            cne_printf("Unsupported flag \'%c'\n", *str);
+            exit(-1);
+        }
+    }
+    return flags;
+}
+
+static int
+process_patterns(struct fwd_info *fwd)
+{
+    char *entries[4];
+    unsigned int id;
+
+    fwd->hs_ids         = (unsigned int *)calloc(fwd->hs_pattern_count + 1, sizeof(unsigned int));
+    fwd->hs_expressions = calloc(fwd->hs_pattern_count + 1, sizeof(char *));
+    fwd->hs_flags       = (unsigned int *)calloc(fwd->hs_pattern_count + 1, sizeof(unsigned int));
+    if (!fwd->hs_expressions || !fwd->hs_flags || !fwd->hs_ids) {
+        free(fwd->hs_ids);
+        free((void *)(uintptr_t)fwd->hs_expressions);
+        free(fwd->hs_flags);
+        return -1;
+    }
+
+    for (int i = 0; i < fwd->hs_pattern_count; i++) {
+        int n = cne_strtok(fwd->hs_patterns[i], "/", entries, cne_countof(entries));
+
+        if (n < 2 || n > 3)
+            CNE_ERR_RET("Hyperscan pattern is invalid\n");
+
+        if (entries[0] == NULL)
```

Comment:
is there a mismatch here between the index in the check `0` and the index in the CNE_ERR_RET statement `i` Or is the intention to print the fwd->hs_patterns[i] in the CNE_ERR_RET statement